app/command/registration.js

```diff
@@ -46,19 +46,23 @@ class Registration {
 		};
 
 		function add( user ) {
-			registered.push( user );
+			// Add the player only if he/she is not registered already.
+			// We could return if user in in registered pool already but it could be abused to check if someone else is already registered.
+			if ( registered.indexOf( user ) === -1 ) {
+				registered.push( user );
```

Comment:
Actually your plan is invalid because of:

https://github.com/pjasiun/fuma/blob/master/app/command/registration.js#L57

Registered will never be > 4 :dancer: (I've wrote test for that).

I'm updating this PR with tests...